objective.py

```diff
@@ -105,26 +107,29 @@ def evaluate_result(self, cv_results, dict_estimators):
                                  sample_domain=self.sample_domain_test
                                  )
 
+        sample_domain_test_source = self.sample_domain_test[extract_source_indices(self.sample_domain_test)]
+        sample_domain_test_target = self.sample_domain_test[~extract_source_indices(self.sample_domain_test)]
+
         all_metrics = {}
 
         for criterion, estimator in dict_estimators.items():
             # TRAIN metrics
             # Source accuracy
-            y_pred_train_source = estimator.predict(X_train_source)
-            y_pred_train_source_proba = estimator.predict_proba(X_train_source)
+            y_pred_train_source = estimator.predict(X_train_source, sample_domain = sample_domain_train_source)
+            y_pred_train_source_proba = estimator.predict_proba(X_train_source, sample_domain = sample_domain_train_source)
 
             # Target accuracy
-            y_pred_train_target = estimator.predict(X_train_target)
-            y_pred_train_target_proba = estimator.predict_proba(X_train_target)
+            y_pred_train_target = estimator.predict(X_train_target, sample_domain = sample_domain_train_target)
+            y_pred_train_target_proba = estimator.predict_proba(X_train_target, sample_domain = sample_domain_train_target)
 
             # TEST metrics
             # Source accuracy
-            y_pred_test_source = estimator.predict(X_test_source)
-            y_pred_test_source_proba = estimator.predict_proba(X_test_source)
+            y_pred_test_source = estimator.predict(X_test_source, sample_domain = sample_domain_test_source)
+            y_pred_test_source_proba = estimator.predict_proba(X_test_source, sample_domain = sample_domain_test_source)
 
             # Target accuracy
-            y_pred_test_target = estimator.predict(X_test_target)
-            y_pred_test_target_proba = estimator.predict_proba(X_test_target)
+            y_pred_test_target = estimator.predict(X_test_target, sample_domain = sample_domain_test_target)
+            y_pred_test_target_proba = estimator.predict_proba(X_test_target, sample_domain = sample_domain_test_target)
```

Comment:
Why must we add all these `sample_domain` since we have not done it before?

Reply:
Tbh, I think it was a mistake to not add all these `sample_domain` from the beginning. We didn't get any error raised before because in the implementation of shallow methods, the `sample_domain` is inferred as `target` if empty, while its not the case with deep methods.
All that to say, I think we should add them